src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -5352,22 +5353,271 @@ void foo()
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
-        public void SquareBracesEmpty_True()
+        public void SpaceWithinEmptyBracketPrecedencesSpaceBeforeOrAfterComma_00()
         {
             var code = @"
 class Program
 {
+    int[ ] x;
+    int[ , ] y;
+    int[ , , ] z = new int[1,2,3];
+    var a = new[ ] { 0 };
+}";
+
+            var expected00 = @"
+class Program
+{
     int[] x;
+    int[,] y;
+    int[,,] z = new int[1,2,3];
+    var a = new[] { 0 };
 }";
 
-            var expected = @"
+            var expected01 = @"
+class Program
+{
+    int[] x;
+    int[,] y;
+    int[,,] z = new int[1, 2, 3];
+    var a = new[] { 0 };
+}";
+
+            var expected10 = @"
+class Program
+{
+    int[] x;
+    int[,] y;
+    int[,,] z = new int[1 ,2 ,3];
+    var a = new[] { 0 };
+}";
+
+            var expected11 = @"
+class Program
+{
+    int[] x;
+    int[,] y;
+    int[,,] z = new int[1 , 2 , 3];
+    var a = new[] { 0 };
+}";
+
+            var options = new Dictionary<OptionKey, object>()
+            {
+                { SpaceBeforeComma, false },
+                { SpaceAfterComma, false },
+                { SpaceBetweenEmptySquareBrackets, false },
+                { SpaceWithinSquareBrackets, false },
+            };
+            AssertFormat("// 00" + expected00, "// 00" +  code, changedOptionSet: options);
+
+            options[SpaceBeforeComma] = true;
+            AssertFormat("// 10" + expected10, "// 10" +  code, changedOptionSet: options);
```

Comment:
The combinations of the testcases are really good. It would be nice if the asserts fall into its own testcase. This way, when something fails, there is just one scenario that is at fault and we dont want to be looking at which assert, among the 4, that failed the whole testcase.

Also a different name instead of `expected00` would be more readable. In this case, it would be something like `expectedWithNoSpaceBeforeCommaAndAfterComma`. When this particular assert fails, the user will automatically know what the expectation is instead of trying to understand the naming convention and finding out what went wrong. Descriptive variable names are always preferred over cryptic names